src/app/calculator/steam/pipe-insulation-reduction/pipe-insulation-reduction.service.ts

```diff
@@ -72,6 +72,13 @@ export class PipeInsulationReductionService {
       heatedOrChilled: [{ value: obj.heatedOrChilled, disabled: !isBaseline }]
     });
 
+    if (obj.heatedOrChilled == 0) {
```

Comment:
For both calculators - you can create a method on this service that adds validators to the form and returns it, based on val or heat or chilled. Use the service method in the form component and remove the logic there.